app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,25 @@
+class SessionsController < ApplicationController
+
+  before_action :require_user, only: [:destroy]
+
+  def new
+    redirect_to home_path if logged_in?
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
```

Comment:
I believe `find_by(email:` is now recommended instead of dynamic finders. Dynamic finders use `method_missing` which makes them inherently slower.